compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIECreateLogicalObjectFifoLink.cpp

```diff
@@ -161,8 +159,30 @@ LogicalResult createLogicalObjectFifoLink(
   // Insert the `LogicalObjectFifoLink` after the last user operation.
   if (lastUserOp) {
     rewriter.setInsertionPointAfter(lastUserOp);
-    rewriter.create<AMDAIE::LogicalObjectFifoLink>(rewriter.getUnknownLoc(),
-                                                   inResults, outResults);
+    auto linkOp = rewriter.create<AMDAIE::LogicalObjectFifoLink>(
+        rewriter.getUnknownLoc(), inResults, outResults);
+    newLinkOps.push_back(linkOp);
+  }
+  return success();
+}
+
+LogicalResult discardLinkNonZeroOffsets(RewriterBase &rewriter,
+                                        AMDAIE::LogicalObjectFifoLink linkOp) {
+  for (Value input : linkOp.getIns()) {
+    if (auto stridedOp = dyn_cast<AMDAIE::DoublyStridedCopyOpInterface>(
+            input.getDefiningOp())) {
```

Comment:
Here and below this should've been `dyn_cast_or_null` because `getDefiningOp()` is being used.

Perhaps can be addressed in the `aie.flow` extraction PR? @jtuyls

Reply:
Thanks for pointing this out. I went through the codebase for all cases and created a separate PR: https://github.com/nod-ai/iree-amd-aie/pull/744

Reply:
Thank you for the amazing cleanup Jorn!